apps/files/src/components/FilesAppBar.vue

```diff
@@ -312,6 +323,14 @@ export default {
       })
 
       return insufficientPermissions === false
+    },
+
+    canCopy() {
+      if (this.publicPage()) {
+        return this.currentFolder.canCreate()
+      }
+
+      return true
```

Comment:
add TODO "more extended permission check" ?

in theory a copy is always allowed, it's the destination that might not allow writing, in which case we might want to display the destination greyed out or differently

you can test this in OC 10 by having a received share folder which is read-only and then try to copy into that one

Reply:
never mind, you have a check already